bpf/bpf_host.c

```diff
@@ -856,6 +862,12 @@ handle_ipv4_cont(struct __ctx_buff *ctx, __u32 secctx, const bool from_host,
 					 info->sec_identity, true, false);
 #endif
 
+#if defined(ENABLE_IPSEC)
+	if (from_proxy &&
+	    (!info || !identity_is_cluster(info->sec_identity)))
+		ctx->mark = MARK_MAGIC_PROXY_TO_WORLD;
+#endif /* ENABLE_IPSEC */
+
 	return CTX_ACT_OK;
```

Comment:
For my understanding - we *could* limit this to `native-routing` (and fold it into the code section above), as we don't install the `0xb00/0xf00 lookup 2005` rule otherwise. Correct?